### PR TITLE
Feature - Unlimited Approval

### PR DESCRIPTION
**Pull request by pingustar**

### CODE REVIEW

**Comment by Velua** [comment]
@pingustar ```onPrompt``` has now been introduced, methods on ```ethBancor.ts``` which use it will await until the ```selectedPromptReceiver$.next``` function is called with the questions ID.

**Comment by pingustar** [comment]
@Velua please review.

I have added the `onPrompt` and updated pretty much all modals. Turned into a fairly big refactor.

One thing I noticed is that `Stake Single Side` -> `Unlimited Approve` after the approve tx I get an error for the stake tx. Would be great if you can have a look at that.

**Comment by Velua** [comment]
Thanks! Good work. Will check it soon

File: src/components/pool/PoolActionsRemoveV1.vue
Comment (Velua) [inline src/components/pool/PoolActionsRemoveV1.vue:174]: Same here, don't we need ```onPrompt``` ?

File: src/components/protection/AddProtectionSingle.vue
Comment (Velua) [inline src/components/protection/AddProtectionSingle.vue:140]: Should we move this to i8ln ?

File: src/components/protection/WithdrawProtectionDouble.vue
Comment (Velua) [inline src/components/protection/WithdrawProtectionDouble.vue:185]: To handle

**Review by Velua — CHANGES_REQUESTED** [review]

File: src/components/protection/AddProtectionSingle.vue
Comment (pingustar) [reply src/components/protection/AddProtectionSingle.vue:140]: ah yeah damn forgot about that. Will create an issue and fix that as a follow up. 

Lets get this out the door if it passes review.

**Review by Velua — APPROVED** [review]
LGTM

**Comment by ashachaf** [comment]
@meytal2021

**Comment by ashachaf** [comment]
@Velua @pingustar 
we are considering introducing a dedicated approve popup
flow change:
current: 
1. setup your swap 
2. confirmation popup 
3. user confirms on popup
4. we trigger approve transaction to metamask  
5. user confirms on metamast
6. we trigger swap transaction to metamask
7. user confirms on metamask

updated:
1. setup your swap 
2. approve confirmation popup 
3. user confirms on popup
4. we trigger approve transaction to metamask  
5. user confirms on metamast
**6. swap confirmation popup**
**7. user confirms on popup**
6. we trigger swap transaction to metamask
7. user confirms on metamask

**Comment by Velua** [comment]
> @Velua @pingustar 
> we are considering introducing a dedicated approve popup
> flow change:
> current: 
> 1. setup your swap 
> 2. confirmation popup 
> 3. user confirms on popup
> 4. we trigger approve transaction to metamask  
> 5. user confirms on metamast
> 6. we trigger swap transaction to metamask
> 7. user confirms on metamask
> 
> updated:
> 1. setup your swap 
> 2. approve confirmation popup 
> 3. user confirms on popup
> 4. we trigger approve transaction to metamask  
> 5. user confirms on metamast
> **6. swap confirmation popup**
> **7. user confirms on popup**
> 6. we trigger swap transaction to metamask
> 7. user confirms on metamask
> 

Not sure I understand what this means, what are we doing differently that the current set up?

**Comment by ashachaf** [comment]
flow change:
1. user goes to swap and click "swap"
2. we check: IF there is no approve, display APPROVE popup https://www.figma.com/file/3yED7o7TgoLwoEjV2EkRLL/Bancor-V2?node-id=2606%3A5275
3. user clicks "approve" and we trigger the approve transaction to his metamask
4. user signs the approve transaction on metamask (during this time, we show the pending popup)
5. once transaction confirms on the blockchain, we trigger the swap confirmation popup https://www.figma.com/file/3yED7o7TgoLwoEjV2EkRLL/Bancor-V2?node-id=2606%3A5275
6. user clicks "confirm" and we trigger the swap transaction to his metamask
7. user signs the swap transaction on metamask (during this time, we show the pending popup)

unlike today's flow where we trigger metamask transactions one after the other with no popup in between.

**Comment by ashachaf** [comment]
please address these things:
1. approve popup 
- title change. current: Approve -> updated: Approve Token
- supporting text update. current: Before you can proceed, you need to approve your spending amount. -> updated: Before you can proceed, you need to approve your token spending amount.

2. confirm token swap popup
- supporting text -> align left. (currently this text is centered)
example: 
image

3. confirm withdraw protected position popup
- update text
current: 
~50% of your protected position
0.185057 BNT
updated:
50% of your protected position
~0.185057 BNT
- align this text to the left
image

4. "successful completion" popup was not updated
5. failed transaction popup was not updated
6. stake tokens to vote process is not using the popups
7. add dual side liquidity
- approve trigger 2 approve transactions (one for each reserve)
- there is no confirm add liquidity popup 
8. add single side liquidity
- there is no "add liquidity" popup if i see the approve popup first

**Comment by pingustar** [comment]
@Velua can you take a look at 7. and 8. please. Seems to be the case for most if not all confirm prompts. The confirm prompt only shows up when the amount is already approved. If the approve screen is first, then right after approve it goes straight to metamask.

Correct flow:
approved? ->  yes -> confirm tx -> metamask
approved? -> no -> approve tx -> confirm tx -> metamask

**Comment by ashachaf** [comment]
> @Velua can you take a look at 7. and 8. please. Seems to be the case for most if not all confirm prompts. The confirm prompt only shows up when the amount is already approved. If the approve screen is first, then right after approve it goes straight to metamask.
> 
> Correct flow:
> approved? -> yes -> confirm modal -> metamask
> approved? -> no -> approve modal -> metamask -> confirm modal -> metamask

adding 2 parts that are also missing.
- point 1: swap approve popup
- [ ] update title from "Approve" to "Approve Token"

- point 6: was not done. when you try to stake vbnt into the staking contract.

**Comment by ashachaf** [comment]
@pingustar 
approve popup:
Instead of “Approve Token”  on the CTA->
“Unlimited Approval”

Instead of “Approve limited permission” on the text link ->
“Limited Approval”

**Review by ashachaf — APPROVED** [review]

**Review by Ran73Cohen — CHANGES_REQUESTED** [review]
A lot of places are missing i18n

File: src/components/modals/ModalTxAction.vue
Comment (Ran73Cohen) [inline src/components/modals/ModalTxAction.vue:39]: ^

File: src/components/modals/ModalTxAction.vue
Comment (Ran73Cohen) [inline src/components/modals/ModalTxAction.vue:58]: ^

File: src/components/modals/ModalTxAction.vue
Comment (Ran73Cohen) [inline src/components/modals/ModalTxAction.vue:51]: ^

**Review by Ran73Cohen — COMMENTED** [review]
Just a handful

**Comment by pingustar** [comment]
@ashachaf I have extracted the remaining parts into separate issues as they are mostly unrelated to the original request.

We can get the unlimited approve feature out the door now and fix these 3 issues as a follow up:

#1161 
#1162 
#1164

**Pull request merged**
